src/provider.py

```diff
@@ -55,11 +52,15 @@ def on_topic_requested(self, event: TopicRequestedEvent):
             event.defer()
             return
 
+        # on all unit update the server properties to enable external listener if needed
+        self.charm._on_config_changed(event)
+
+        if not self.charm.unit.is_leader():
+            return
+
```

Comment:
**issue:** I don't *think* this is needed, because `on_topic_requested` means that we have a new relation, and by extension a new username/password.
On L77 and L90 we update relation data, which indirectly triggers a `config_changed` event, so the extra call isn't needed, right?

Or is this to cover the case where a related application changes their requested `topic`? Does that still call `on_topic_requested`?

If so, we should probably just add that new thing to relation data, and let `on_config_changed` be called indirectly as before.

Reply:
In this case, I want the update of server configuration before setting the relation data.